pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,135 @@
-        cls.tags_api = ContentTagsApi(client_api)
-        cls.versions_api = RepositoriesContainerVersionsApi(client_api)
+    assert repository.latest_version_href
 
-        cls.repository = None
+    latest_version_href = repository.latest_version_href
+    sync(
+        container_repo, container_remote
+    )  # We expect that this second sync doesn't create a new repo version
+    repository = container_repository_api.read(repository.pulp_href)
 
-    def test_sync_with_non_existing_included_tag(self):
-        """Check whether the sync machinery ignores a non-existing tag."""
-        include_tags = ["manifest_a", "non_existing_manifest"]
-        self.sync_repository_with_filtered_tags(include_tags)
+    assert repository.latest_version_href == latest_version_href
 
-        self.assert_synced_tags(["manifest_a"])
 
-    def test_sync_with_included_tags(self):
-        """Test whether the repository is synced only with included tags."""
-        include_tags = ["manifest_a", "manifest_b", "manifest_c"]
-        self.sync_repository_with_filtered_tags(include_tags)
+@pytest.mark.parallel
+def test_sync_invalid_url(container_repo, container_remote_api, sync, gen_object_with_cleanup):
+    remote_data = gen_container_remote(url="http://i-am-an-invalid-url.com/invalid/")
+    container_remote = gen_object_with_cleanup(
+        container_remote_api, ContainerContainerRemote(**remote_data)
+    )
 
-        self.assert_synced_tags(include_tags)
+    with pytest.raises(PulpTaskError) as ctx:
+        sync(container_repo, container_remote)
 
-    def test_sync_with_included_tags_using_wildcard(self):
-        """Test whether the repository is synced only with included tags that use wildcards."""
-        include_tags = [
-            "ml_iv",
-            "ml_ii",
-            "manifest_a",
-            "manifest_b",
-            "manifest_c",
-            "manifest_d",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(["ml_??", "manifest*"])
-
-        self.assert_synced_tags(include_tags)
-
-    def test_sync_with_complex_filtering(self):
-        """Test sync repository with included and excluded that use wildcards."""
-        include_tags = [
-            "manifest_a",
-            "manifest_c",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(
-            include_tags=["manifest_*"], exclude_tags=["*_[bd]"]
-        )
+    assert "[Name or service not known]" in ctx.value.task.error["description"]
 
-        self.assert_synced_tags(include_tags)
 
-    def sync_repository_with_filtered_tags(self, include_tags=None, exclude_tags=None):
-        """Sync a new repository with the included tags passed as an argument."""
-        self.repository = self.repositories_api.create(ContainerContainerRepository(**gen_repo()))
-        self.addCleanup(self.repositories_api.delete, self.repository.pulp_href)
+@pytest.mark.parallel
+def test_sync_idempotency(
```

Comment:
Can this test be merged with `test_basic_sync`? We will only have an additional assertion on top of it.
```
    assert first_sync_tags_named_a.count == 1
    assert second_sync_tags_named_a.count == 1
```